dotnet/src/Agents/OpenAI/Internal/OpenAIClientFactory.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.ClientModel.Primitives;
+using System.Net.Http;
+using System.Threading;
+using Azure.AI.OpenAI;
+using Microsoft.SemanticKernel.Http;
+using OpenAI;
+
+namespace Microsoft.SemanticKernel.Agents.OpenAI.Internal;
+
+/// <summary>
+/// Factory for creating <see cref="OpenAIClient"/>.
+/// </summary>
+internal static class OpenAIClientFactory
+{
+    /// <summary>
+    /// Avoids an exception from OpenAI Client when a custom endpoint is provided without an API key.
+    /// </summary>
+    private const string SingleSpaceKey = " ";
+
+    /// <summary>
+    /// Creates an OpenAI client based on the provided configuration.
+    /// </summary>
+    /// <param name="config">Configuration required to target a specific Open AI service</param>
+    /// <returns>An initialized Open AI client</returns>
+    public static OpenAIClient CreateClient(OpenAIServiceConfiguration config)
+    {
+        // Inspect options
+        switch (config.Type)
+        {
+            case OpenAIServiceConfiguration.OpenAIServiceType.AzureOpenAI:
+            {
+                AzureOpenAIClientOptions clientOptions = CreateAzureClientOptions(config);
+
+                if (config.Credential is not null)
+                {
+                    return new AzureOpenAIClient(config.Endpoint, config.Credential, clientOptions);
+                }
+
+                if (!string.IsNullOrEmpty(config.ApiKey))
+                {
+                    return new AzureOpenAIClient(config.Endpoint, config.ApiKey!, clientOptions);
+                }
+
+                throw new KernelException($"Unsupported configuration type: {config.Type}");
+            }
+            case OpenAIServiceConfiguration.OpenAIServiceType.OpenAI:
+            {
+                OpenAIClientOptions clientOptions = CreateOpenAIClientOptions(config);
+                return new OpenAIClient(config.ApiKey ?? SingleSpaceKey, clientOptions);
```

Comment:
Is it OK to use the OpenAIClient for non OpenAI endpoints?

Reply:
I suppose it depends on how we defined a non-Open AI endpoint.  A local proxy might ultimately resolve to an Open AI endpoint.

This was intended to align with the connector logic:

https://github.com/microsoft/semantic-kernel/blob/feature-connectors-openai/dotnet/src/Connectors/Connectors.OpenAIV2/Core/ClientCore.cs#L119